components/voting.tsx

```diff
@@ -386,101 +417,140 @@ export default class Voting extends React.PureComponent<VotingProps, VotingState
-                    )}
-                    {this.isInShortlist(s) && (
-                      <span
-                        className="fa fa-list-ol status"
-                        aria-label="Shortlisted"
-                        role="status"
-                        title="Shortlisted"
-                      />
-                    )}
-                    {this.isFlagged(s) && (
-                      <span className="fa fa-flag status" aria-label="Flag" role="status" title="Flagged" />
-                    )}
-                    {s.Title}
-                    <br />
-                    {(s.Tags || []).map(tag => (
-                      <React.Fragment key={tag}>
-                        <span className="badge">{tag}</span>{' '}
-                      </React.Fragment>
-                    ))}
-                    <small
-                      style={{
-                        display: 'block',
-                        marginTop: '10px',
-                        visibility: this.state.expandAll ? 'hidden' : 'visible',
-                      }}
-                    >
-                      <span className="fa fa-plus" title="More details" /> Tap for session details
-                    </small>
-                    {!this.state.submitted && (
-                      <div style={{ textAlign: 'right', paddingTop: '10px' }}>
-                        <button
-                          onClick={e => {
-                            this.toggleFlagged(s)
-                            e.stopPropagation()
-                            e.preventDefault()
-                          }}
-                          className="btn btn-secondary btn-sm"
-                        >
-                          {!this.isFlagged(s) ? 'Flag' : 'Un-flag'}
-                        </button>{' '}
-                        <button
-                          onClick={e => {
-                            this.toggleShortlist(s)
-                            e.stopPropagation()
-                            e.preventDefault()
-                          }}
-                          className="btn btn-secondary btn-sm"
-                        >
-                          {!this.isInShortlist(s) ? 'Shortlist' : 'Un-shortlist'}
-                        </button>{' '}
-                        <button
-                          onClick={e => {
-                            this.toggleVote(s)
-                            e.stopPropagation()
-                            e.preventDefault()
-                          }}
-                          className="btn btn-primary btn-sm"
-                          disabled={this.state.votes.length >= this.props.maxVotes && !this.isVotedFor(s)}
+        <DragDropContext
+          onDragEnd={(result: DropResult) => {
+            this.onDragEnd(result)
+          }}
+        >
+          <Droppable droppableId="voteDroppable">
+            {provider => (
+              <div ref={provider.innerRef} {...provider.droppableProps}>
```

Comment:
Had trouble adding ref to PanelGroup, hence the wrapping div. This needs reworking the semantics are all off.